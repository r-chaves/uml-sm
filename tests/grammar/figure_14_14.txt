stm uml_figure_14_14
{
    initial -> verifyCard
    state verifyCard
    submachine readAmount:ReadAmountSM
    {
        exitPoint aborted
    }

    readAmount.aborted -> CardReleased
    state CardReleased
    readAmount -> outOfService / outOfService
    readAmount -> verifyTransaction
    state outOfService
    state verifyTransaction
    verifyTransaction -> CardReleased / releaseCard
}
