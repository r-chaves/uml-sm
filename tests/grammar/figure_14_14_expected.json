{
  "stm_name": "uml_figure_14_14", 
  "region_member": [
    {
      "src": "initial", 
      "guard": null, 
      "trigger": null, 
      "target": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "verifyCard"
      }, 
      "effect": null
    }, 
    {
      "region_member": null, 
      "region": null, 
      "name": "verifyCard", 
      "behavior": null
    }, 
    {
      "exit_point": "aborted", 
      "entry_point": null, 
      "name": {
        "state": "readAmount", 
        "submachine": "ReadAmountSM"
      }
    }, 
    {
      "src": {
        "scope": null, 
        "region": null, 
        "conn_point": "aborted", 
        "name": "readAmount"
      }, 
      "guard": null, 
      "trigger": null, 
      "target": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "CardReleased"
      }, 
      "effect": null
    }, 
    {
      "region_member": null, 
      "region": null, 
      "name": "CardReleased", 
      "behavior": null
    }, 
    {
      "src": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "readAmount"
      }, 
      "guard": null, 
      "trigger": null, 
      "target": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "outOfService"
      }, 
      "effect": "outOfService"
    }, 
    {
      "src": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "readAmount"
      }, 
      "guard": null, 
      "trigger": null, 
      "target": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "verifyTransaction"
      }, 
      "effect": null
    }, 
    {
      "region_member": null, 
      "region": null, 
      "name": "outOfService", 
      "behavior": null
    }, 
    {
      "region_member": null, 
      "region": null, 
      "name": "verifyTransaction", 
      "behavior": null
    }, 
    {
      "src": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "verifyTransaction"
      }, 
      "guard": null, 
      "trigger": null, 
      "target": {
        "scope": null, 
        "region": null, 
        "conn_point": null, 
        "name": "CardReleased"
      }, 
      "effect": "releaseCard"
    }
  ]
}
